install.py

```diff
@@ -226,20 +222,18 @@ def validate_path(path):
     cuda_dir = validate_path(cuda_dir)
     nccl_dir = validate_path(nccl_dir)
     tblis_dir = validate_path(tblis_dir)
-    legate_dir = validate_path(legate_dir)
     thrust_dir = validate_path(thrust_dir)
     curand_dir = validate_path(curand_dir)
     gasnet_dir = validate_path(gasnet_dir)
     cutensor_dir = validate_path(cutensor_dir)
     openblas_dir = validate_path(openblas_dir)
 
-    if legate_dir is None:
-        try:
-            import legate.install_info as lg_install_info
+    try:
+        import legate.install_info as lg_install_info
 
-            legate_dir = dirname(lg_install_info.libpath)
-        except Exception:
-            pass
+        legate_dir = dirname(lg_install_info.libpath)
+    except Exception:
+        legate_dir = None
```

Comment:
Copying from previous iteration of this PR:

> My only question is should this be an error instead?
 
It looks like cmake exposes the capability to build legate.core as part of building cunumeric (see options legate_branch and legate_url), so I assume leaving legate_dir undefined (and failing to find legate.install_info) was meant to trigger this behavior?

However, I recall reading that (at least currently) a recursive build of legate.core through cunumeric will not do a complete enough installation (e.g. legion_python and the legate driver will not get installed). Therefore it might be a good idea to require that legate.install_info is found at this stage (and correspondingly remove legate_url and legate_branch)?

@trxcllnt do I have it right?

Reply:
Yeah, we do need the Python modules to be installed (in either editable mode or not) to get an entire working setup. CMake can build/install all the C++ libraries in one command (because of CPM), but Python doesn't have a way to say, "find or install this other module as part of pip-installing the current module."

We work around this by having legate.core's `install.py` manually invoke `pip install` for Legion. We could do the same thing in cuNumeric's `install.py` (invoke `pip install` for legate.core and Legion if we built them as side-effects of building cuNumeric), but we aren't presently.

Reply:
2 sounds good to me